src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java

```diff
@@ -235,4 +238,23 @@ public void updateOnTheFlySecurityHotspots(@NotNull Map<VirtualFile, Collection<
       content.setDisplayName(buildTabName(count, SonarLintToolWindowFactory.SECURITY_HOTSPOTS_TAB_TITLE));
     }
   }
+
+  public boolean isSecurityHotspotsTabActive() {
+    var toolWindow = getToolWindow();
+    if (toolWindow != null && securityHotspotsContent != null) {
+      return toolWindow.isVisible() && securityHotspotsContent.isSelected();
+    }
+    return false;
+  }
+
+  @Override
+  public void selectionChanged(@NotNull ContentManagerEvent event) {
+    var component = event.getContent().getComponent();
+    if (component instanceof SecurityHotspotsPanel) {
```

Comment:
I think we also need to clear the annotations when leaving the security hotspots tab